src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/ExportClient/OtlpRetry.cs

```diff
@@ -0,0 +1,248 @@
+    public const int InitialBackoffMilliseconds = 1000;
+    private const int MaxBackoffMilliseconds = 5000;
+    private const double BackoffMultiplier = 1.5;
+
+#if !NET6_0_OR_GREATER
+    private static readonly Random Random = new Random();
+#endif
+
+    public static bool TryGetHttpRetryResult(HttpStatusCode statusCode, DateTime? deadline, HttpResponseHeaders responseHeaders, int retryDelayMilliseconds, out RetryResult retryResult)
+    {
+        return OtlpRetry.TryGetRetryResult(statusCode, IsHttpStatusCodeRetryable, deadline, responseHeaders, TryGetHttpRetryDelay, retryDelayMilliseconds, out retryResult);
+    }
+
+    public static bool TryGetGrpcRetryResult(StatusCode statusCode, DateTime? deadline, Metadata trailers, int retryDelayMilliseconds, out RetryResult retryResult)
+    {
+        return OtlpRetry.TryGetRetryResult(statusCode, IsGrpcStatusCodeRetryable, deadline, trailers, TryGetGrpcRetryDelay, retryDelayMilliseconds, out retryResult);
+    }
+
+    private static bool TryGetRetryResult<TStatusCode, TCarrier>(TStatusCode statusCode, Func<TStatusCode, bool, bool> isRetryable, DateTime? deadline, TCarrier carrier, Func<TStatusCode, TCarrier, TimeSpan?> throttleGetter, int nextRetryDelayMilliseconds, out RetryResult retryResult)
+    {
+        retryResult = default;
+
+        // TODO: Consider introducing a fixed max number of retries (e.g. max 5 retries).
+        // The spec does not specify a max number of retries, but it may be bad to not cap the number of attempts.
+        // Without a max number of retry attempts, retries would continue until the deadline.
+        // Maybe this is ok? However, it may lead to an unexpected behavioral change. For example:
+        //    1) When using a batch processor, a longer delay due to repeated
+        //       retries up to the deadline may lead to a higher chance that the queue will be exhausted.
+        //    2) When using the simple processor, a longer delay due to repeated
+        //       retries up to the deadline will lead to a prolonged blocking call.
+        // if (attemptCount >= MaxAttempts)
+        // {
+        //     return false
+        // }
+
+        if (IsDeadlineExceeded(deadline))
+        {
+            return false;
+        }
+
+        var throttleDelay = throttleGetter(statusCode, carrier);
+        var retryable = isRetryable(statusCode, throttleDelay.HasValue);
+        if (!retryable)
+        {
+            return false;
+        }
+
+        var delayDuration = throttleDelay.HasValue
+            ? throttleDelay.Value
+            : TimeSpan.FromMilliseconds(GetRandomNumber(0, nextRetryDelayMilliseconds));
+
+        if (deadline.HasValue && IsDeadlineExceeded(deadline + delayDuration))
+        {
+            return false;
+        }
+
+        if (throttleDelay.HasValue)
+        {
+            try
+            {
+                nextRetryDelayMilliseconds = Convert.ToInt32(throttleDelay.Value.TotalMilliseconds);
```

Comment:
Could we just use `double` as the type for `InitialBackoffMilliseconds` and `nextRetryDelayMilliseconds` to keep things simple?

Reply:
I'd considered this but landed on converting for now thinking it might simplify the consuming logic. Though, I might become convinced using a double is the better way to go. I'm happy to come back and reconsider this later if that works for you. Added a TODO.